From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: quh4gko8 <88831734+quh4gko8@users.noreply.github.com>
Date: Thu, 28 May 2026 15:49:33 +0000
Subject: [PATCH] Add workaround for adding site settings exception for Motion
 Sensors

_fixup! Initial support for toggling Sensors site settings per-site
---
 .../site_settings/SensorsSettingsHooks.java   | 33 +++++++++++++++++++
 1 file changed, 33 insertions(+)

diff --git a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SensorsSettingsHooks.java b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SensorsSettingsHooks.java
index 5ebbc140a26c2..d208aff34ab1a 100644
--- a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SensorsSettingsHooks.java
+++ b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SensorsSettingsHooks.java
@@ -14,12 +14,16 @@ import org.chromium.build.annotations.Nullable;
 import org.chromium.components.content_settings.ContentSetting;
 import org.chromium.components.content_settings.ContentSettingsType;
 import org.chromium.content_public.browser.BrowserContextHandle;
+import org.chromium.content_public.common.ContentUrlConstants;
 
 import app.vanadium.config.VanadiumConfParser;
 import app.vanadium.ext.CustomOSPermissions;
 
 import static app.vanadium.config.proto.VanadiumConfigProto.Flag.FlagType.DISABLED_FEATURE;
 
+import java.util.ArrayList;
+import java.util.Collection;
+
 @NullMarked
 public final class SensorsSettingsHooks {
 
@@ -98,6 +102,35 @@ public final class SensorsSettingsHooks {
             return false; // TODO: support with improved exception handling
         }
 
+        @Override
+        protected Collection<Website> applyFilters(Collection<Website> sites) {
+            String httpsScheme = ContentUrlConstants.HTTPS_SCHEME + "://";
+            Collection<Website> filtered = new ArrayList<>();
+            for (Website site : sites) {
+                if (site.getAddress().getOrigin().startsWith(httpsScheme)) {
+                    filtered.add(site);
+                }
+            }
+            return filtered;
+        }
+
+        @Override
+        protected @Nullable String maybeModifyPrimaryPattern(String pattern) {
+            String httpsScheme = ContentUrlConstants.HTTPS_SCHEME + "://";
+            String httpsPort = ":443";
+            String patternNoScheme;
+            if (pattern.startsWith(httpsScheme)) {
+                patternNoScheme = pattern.substring(pattern.indexOf(httpsScheme));
+            } else {
+                patternNoScheme = pattern;
+            }
+            if (patternNoScheme.contains(":")) {
+                return httpsScheme + pattern.substring(0, pattern.indexOf(":")) + httpsPort;
+
+            }
+            return httpsScheme + pattern + httpsPort;
+        }
+
         @Override
         protected int getAddExceptionDialogMessageResourceId(boolean isCategoryEnabled) {
             return isCategoryEnabled ? R.string.website_settings_add_site_description_motion_sensors_block
